spec/models/manageiq/providers/openstack/cloud_manager/template_spec.rb

```diff
@@ -1,8 +1,43 @@
 describe ManageIQ::Providers::Openstack::CloudManager::Template do
   let(:ems) { FactoryGirl.create(:ems_openstack_with_authentication) }
+  let(:image_attributes) { {:name => "image", :ram => "1"} }
   let(:template_openstack) { FactoryGirl.create :template_openstack, :ext_management_system => ems, :ems_ref => 'one_id' }
   let(:service) { double }
 
+  context 'when create_image' do
+    before do
+      allow(ExtManagementSystem).to receive(:find).with(ems.id).and_return(ems)
+      allow(ems).to receive(:with_provider_connection).with(:service => 'Compute').and_yield(service)
+      allow(service).to receive(:images).and_return(images)
```

Comment:
Some of these lines are not needed.  E.g. ```ExtManagementSystem``` exists in database so it can be found directly from model (created by line 2 and first usage of ```ems``` variable). For example take a look at https://gist.github.com/aufi/16acd1f29086eeb211ceabd3bc98625e I don't want push you to rewrite your code much, just for information.